test/parallel/test-inspector-connect-main-thread.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+const common = require('../common');
+
+common.skipIfInspectorDisabled();
+common.skipIfWorker();
+
+const assert = require('assert');
+const { Session } = require('inspector');
+const path = require('path');
+const { pathToFileURL } = require('url');
+const { isMainThread, parentPort, Worker } = require('worker_threads');
+
+function toDebug() {
+  let a = 1;
+  a = a + 1;
+  return a * 200;
+}
+
+const messagesSent = [];
+
+async function post(session, method, params) {
+  return new Promise((resolve, reject) => {
+    session.post(method, params, (error, success) => {
+      messagesSent.push(method);
+      if (error) {
+        console.log(`Message ${method} produced an error`);
+        reject(error);
+      } else {
+        console.log(`Message ${method} was sent`);
+        resolve(success);
+      }
+    });
+  });
+}
+
+async function waitForNotification(session, notification) {
+  return new Promise((resolve) => session.once(notification, resolve));
+}
+
+function WaitForWorker() {
+  const worker = new Worker(__filename);
+  return new Promise(
+    (resolve) => worker.once('message', () => resolve(worker)));
+}
+
+function waitForWorkerMessages(worker) {
+  let workerDoneResolve, delayedConsoleLogResolve;
+  worker.on('message', ({ consoleLogIn10ms, messagesSent }) => {
+    if (consoleLogIn10ms) {
+      delayedConsoleLogResolve();
+    } else {
+      workerDoneResolve(messagesSent);
+    }
+  });
+  return [
+    new Promise((resolve) => { delayedConsoleLogResolve = resolve; }),
+    new Promise((resolve) => { workerDoneResolve = resolve; })
+  ];
+}
+
+function doConsoleLogIn5ms() {
+  setTimeout(() => {
+    console.log('Message for a test');
+  }, 5);
```

Comment:
Is this used somewhere?

Reply:
Yes, it is for `ensureListenerDoesNotInterrupt`